controllers/warehouse-controller.js

```diff
@@ -111,9 +112,66 @@ const getInventoriesOfWarehouseById = async (req, res) => {
   }
 };
 
+const editWarehouseById = async (req, res) => {
+
+  const isWhIdValid = await isWarehouseIdValid(req.params.id);
+  if (!isWhIdValid) {
+    return res.status(400).json({
+      message: `Warehouse ID not found`,
+    });
+  }
+
+  const {
+    warehouse_name,
+    address,
+    city,
+    country,
+    contact_name,
+    contact_position,
+    contact_phone,
+    contact_email
+  } = req.body
+
+  if (
+    !warehouse_name ||
+    !address ||
+    !city ||
+    !country ||
+    !contact_name ||
+    !contact_position ||
+    !contact_phone ||
+    !contact_email) {
+    return res.status(400).json({
+      message: `Missing properties in the request body`,
+    })
```

Comment:
Hi Ryan! Your validation logic in editWarehouseById checks for the presence of all required fields, which aligns with the ticket requirements. However, it doesn't validate the format of the phone number and email address.

Reply:
hi, email will be validate on the FE with form input, I'll add sth for the phone, thanks